tests/tests/functional/chart_test.py

```diff
@@ -0,0 +1,196 @@
+
+sys.path.append('./scripts/src/')
+from report import report_info
+
+import yaml
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+
+
+@scenario(
+    "features/chart_good.feature",
+    "A chart provider verifies their chart using the chart verifier docker image"
+)
+def test_chart_source():
+    pass
+
+@given(parsers.parse("I would like to use the <type> profile"),target_fixture="profile_type")
+def profile_type(type):
+    return type
+
+@given(parsers.parse("I will provide a <location> of a <helm_chart>"),target_fixture="chart_location")
+def chart_location(location,helm_chart):
+    return os.path.join(location,helm_chart)
+
+@given(parsers.parse("I will provide a <location> of an expected <report_info>"),target_fixture="report_info_location")
+def report_info_location(location,report_info):
+    return os.path.join(location,report_info)
+
+@when(parsers.parse("I run the <image> chart-verifier verify command against the chart to generate a report"),target_fixture="run_verifier")
+def run_verifier(image, profile_type, chart_location):
+    print(f"\nrun verifier image: {image}, with profile : {profile_type}, and chart: {chart_location}")
+
+    client = docker.from_env()
+    verifier_image_name = "quay.io/redhat-certification/chart-verifier"
+
+    image_tag = os.environ.get("VERIFIER_IMAGE_TAG")
+    if not image_tag:
+        image_tag = image
+        if not image_tag:
+           image_tag = "latest"
+        try:
+            verifier_image=client.images.pull(verifier_image_name,tag=image_tag)
+        except docker.errors.APIError as exc:
+            print(f'Error from docker loading image: {verifier_image_name}:{image_tag}')
+            return f"FAIL pulling image : docker.errors.APIError: {exc.args}"
+    else:
+        try:
+            verifier_image = client.images.get(f"{verifier_image_name}:{image_tag}")
+        except docker.errors.ImageNotFound:
+            return f"FAIL image not found:  : {verifier_image_name}:{image_tag}"
+        except docker.errors.APIError as exc:
+            return f"FAIL getting image : {verifier_image_name}:{image_tag} : docker.errors.APIError: {exc.args}"
+
+    os.environ["VERIFIER_IMAGE"] = f"{verifier_image_name}:{image_tag}"
+
+    docker_command = "verify "
+    local_chart = False
+    if chart_location.startswith('http:/') or chart_location.startswith('https:/'):
+        docker_command = docker_command + chart_location
+    else:
+        if os.path.exists(chart_location):
+            docker_command = docker_command + f"/charts/{os.path.basename(chart_location)}"
+            local_chart = True
+        else:
+            return f"FAIL: chart does not exist: {os.path.abspath(chart_location)}"
+
+    if profile_type:
+        docker_command = docker_command + " --set profile.vendorType=" + profile_type
+
+    print(f'docker command: {docker_command}')
+
+    client = docker.from_env()
```

Comment:
Can we can eliminate this? As "client" is already initialized in line no 43.

Reply:
eliminated, thanks.